internal/controllers/topology/cluster/blueprint.go

```diff
@@ -92,5 +93,30 @@ func (r *Reconciler) getBlueprint(ctx context.Context, cluster *clusterv1.Cluste
 		blueprint.MachineDeployments[machineDeploymentClass.Class] = machineDeploymentBlueprint
 	}
 
+	// Loop over the machine pool classes in ClusterClass
+	// and fetch the related templates.
+	for _, machinePoolClass := range blueprint.ClusterClass.Spec.Workers.MachinePools {
+		machinePoolBlueprint := &scope.MachinePoolBlueprint{}
+
+		// Make sure to copy the metadata from the blueprint, which is later layered
+		// with the additional metadata defined in the Cluster's topology section
+		// for the MachinePool that is created or updated.
+		machinePoolClass.Template.Metadata.DeepCopyInto(&machinePoolBlueprint.Metadata)
+
+		// Get the InfrastructureMachinePoolTemplate.
+		machinePoolBlueprint.InfrastructureMachinePoolTemplate, err = r.getReference(ctx, machinePoolClass.Template.Infrastructure.Ref)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get InfrastructureMachinePoolTemplate for %s, MachinePool class %q", tlog.KObj{Obj: blueprint.ClusterClass}, machinePoolClass.Class)
+		}
+
+		// Get the bootstrap machine template.
+		machinePoolBlueprint.BootstrapTemplate, err = r.getReference(ctx, machinePoolClass.Template.Bootstrap.Ref)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get bootstrap config template for %s, MachinePool class %q", tlog.KObj{Obj: blueprint.ClusterClass}, machinePoolClass.Class)
```

Comment:
in the case of a MachinePool, there is no "bootstrap config template", just a bootstrap config (that is shared between all the machines in the pool) we might want to change the wording here (code comment line 112, variable name on list 113, and log error message line 115) to remove the word "template" so it's not confusing

Reply:
I'll have to check the code, but I'm not sure I agree. This code is about getting the referenced object in the ClusterClass. I think in the ClusterClass we would have a KubeadmConfigTemplate as we don't want a single KubeadmConfig for all Clusters of a ClusterClass